### PR TITLE
Refactor: Typos, error handling, slice declaration, strings, parentheses, error comparison, commas, naming, unused params, switch cases, receiver alignment, escape removal.

### PR DESCRIPTION
**Pull request by mstgnz**
Hi Team,

This pull request includes a series of corrections in the project. The changes made are as follows:

- Typos have been corrected.
- Unhandled errors have been addressed and added TODO
- An issue with an empty slice declaration has been fixed.
- Error strings have been normalized.
- Redundant parentheses have been removed.
- Type assertion and equality comparison issues with errors have been resolved.
- Missing commas after conjunctive/linking adverbs have been added.
- Variable naming conflicts have been addressed.
- Unused parameters have been removed.
- There are missing 'default' statements for 'iota' constants in 'switch'. TODO added.
- Receiver names have been aligned.
- Redundant character escapes and type conversions have been eliminated.

Although these fixes did not affect the code execution, the tests were run, the application was stood up and tested.

I hope these changes improve the code quality and readability. Thank you for your review and feedback.

### CODE REVIEW

**Comment by ganigeorgiev** [comment]
Thank you for spending your time on this but this type of changes are not really welcomed as I don't really see much point of reviewing 140+ files.

There is no need for extra TODOs for the statements where the errors are not really needed.
Renaming variable names like `max` to `mx` also doesn't seem right to me as the first seems more readable.

Some of the suggestions like the typos and grammar improvements, the err.Is checks, etc. can be applied and I'll consider them for a future release, but for now I'll close this PR.

**Comment by mstgnz** [comment]
@ganigeorgiev The reason for adding TODOs to places that need to be error handled is to make it easier to go through and not to lose places. If it really will not be handled, it can remain as _ =.

Variables named "max" and "new" are used and there is a warning for these variables. warning: "Variable 'max' collides with the 'builtin' function" so these variable names were changed. they were conflicting variables. there were even variables with the same name as the package name in some places, I did not fix them.

**Comment by ganigeorgiev** [comment]
I can run the linter on my own. As mentioned some of the suggested changes I'll consider applying in a future release, but I don't want to spend time reviewing so many updates without actually having any net positive over the code execution.

**Comment by mstgnz** [comment]
Of course it's your decision, I just fixed the bugs that the editor warned me about when I opened the project with Goland.
These changes do not prevent the code from working, I ran the tests, stood up the application and used it. there are no changes. since I made these corrections in the whole project, it can be perceived as a very big change. if I did it piece by piece, there would be a lot of pull requests.
I fixed it to maybe help. the decision is yours. wish you good work.

**Comment by ganigeorgiev** [comment]
Just because some editor suggested the change doesn't mean that it is always correct or even needed. For example, I strongly disagree renaming the function/method parameters to `_` even when the arguments are not used in the body, because the names of the parameters are part of the function/method declaration. Similarly, in a lot of places the use of parenthesis is also deliberately used to help with the readability.

In any case, sorry if you consider this a waste of time, but I have plenty on my hands and I don't really consider any of the suggested changes really that important or needed.

**Comment by mstgnz** [comment]
it doesn't matter to me, the project is yours :)

File: apis/backup.go [hunk 1]
```diff
@@ -130,7 +134,10 @@ func (api *backupApi) download(c echo.Context) error {
 	if err != nil {
 		return NewBadRequestError("Failed to load backups filesystem.", err)
 	}
-	defer fsys.Close()
```
Comment (mstgnz) [inline apis/backup.go:133]: If the editor does not give a warning for this type of use, it is not clear that an error message has not been handled.

File: daos/view.go [hunk 2]
```diff
@@ -66,7 +66,8 @@ func (dao *Dao) SaveView(name string, selectQuery string) error {
 		if _, err := txDao.TableInfo(name); err != nil {
 			// manually cleanup previously created view in case the func
 			// is called in a nested transaction and the error is discarded
-			txDao.DeleteView(name)
```
Comment (mstgnz) [inline daos/view.go:69]: If the editor does not give a warning for this type of use, it is not clear that an error message has not been handled.

File: daos/view.go [hunk 3]
```diff
@@ -66,7 +66,8 @@ func (dao *Dao) SaveView(name string, selectQuery string) error {
...
+			// TODO implement error
+			_ = txDao.DeleteView(name)
```
Comment (mstgnz) [inline daos/view.go:70]: this way at least it is understood that the error has been bypassed.

File: tools/rest/excerpt_modifier.go [hunk 4]
```diff
@@ -46,8 +46,8 @@ func newExcerptModifier(args ...string) (*excerptModifier, error) {
 		return nil, errors.New("too many arguments - expected (max, withEllipsis?)")
 	}
 
-	max := cast.ToInt(args[0])
```
Comment (mstgnz) [inline tools/rest/excerpt_modifier.go:49]: Variable 'max' collides with the 'builtin' function

File: tools/rest/excerpt_modifier.go [hunk 5]
```diff
@@ -46,8 +46,8 @@ func newExcerptModifier(args ...string) (*excerptModifier, error) {
...
-	if max == 0 {
+	mx := cast.ToInt(args[0])
```
Comment (mstgnz) [inline tools/rest/excerpt_modifier.go:49]: changed to avoid overlap

File: daos/base.go [hunk 6]
```diff
@@ -87,16 +87,16 @@ func (dao *Dao) Clone() *Dao {
 // WithoutHooks returns a new Dao with the same configuration options
 // as the current one, but without create/update/delete hooks.
 func (dao *Dao) WithoutHooks() *Dao {
-	new := dao.Clone()
```
Comment (mstgnz) [inline daos/base.go:90]: Variable 'new' collides with the 'builtin' function

File: daos/base.go [hunk 7]
```diff
@@ -87,16 +87,16 @@ func (dao *Dao) Clone() *Dao {
...
+	clone := dao.Clone()
```
Comment (mstgnz) [inline daos/base.go:90]: changed to avoid overlap

File: daos/admin_test.go [hunk 8]
```diff
@@ -267,6 +267,7 @@ func TestSaveAdmin(t *testing.T) {
 	if saveErr2 != nil {
 		t.Fatal(saveErr2)
 	}
+	// TODO implement error
```
Comment (mstgnz) [inline daos/admin_test.go:270]: TODO added so you can quickly find places where bugs are bypassed.

**Review by mstgnz — COMMENTED** [review]
It was a pull request consisting of changing what I added as a comment in more than one place.

**Comment by ganigeorgiev** [comment]
I understand the changes, there is no need to add a comment to each line.

As mentioned, some of the suggestions I'll consider applying but again I don't want to spend my time reviewing and applying needless changes just because some random internet stranger decided to nitpick.

**Pull request closed without merging**